### PR TITLE
Algolia search category

### PR DESCRIPTION
**Pull request by kohchihao**
In this PR,
- Updated Searchbar to use `Enter` and search. 
  - Hitting enter will bring you to the `All wishes/donations` page.
  - Now a `q=` URL query will be added when you hit the enter button.
- Updated `/wishes/[categoryId]` page to use algolia 
- Updated `/donations/[categoryId]` page to use algolia 
- Sorting that is available
  - `wishes`: newest - oldest, NPO name A-Z, NPO name Z-A
  - `donations`: newest - oldest

Future considerations
- Adding filters in the future will be easy with a few lines of code and defining some attributes on `Algolia`

Cost implications:
- https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/
- https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/in-depth/replicas-impact-on-pricing/

Resources Used:
- https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/
- https://www.algolia.com/doc/api-reference/widgets/sort-by/react/#connector
- https://www.algolia.com/doc/guides/managing-results/refine-results/sorting/in-depth/replicas-impact-on-pricing/
- https://www.algolia.com/doc/api-reference/widgets/configure/react/#examples
- https://www.algolia.com/doc/api-reference/search-api-parameters/
- https://www.algolia.com/doc/api-reference/widgets/infinite-hits/react/#create-a-react-component

### CODE REVIEW

File: src/components/search/modules/NavSearchBar.js [hunk 1]
```diff
@@ -1,77 +1,18 @@
...
-        content={({ position, nudgedLeft, nudgedTop, targetRect, popoverRect }) => {
-          return (
-            <ContentContainer>
-              <>
-                <Index indexName="wishes">
-                  <Container>
-                    <Text weight="bold">Wishes</Text>
-                  </Container>
+  const onEnterPressed = (query, selectedIndex) => {
+    router.push(`/${selectedIndex}/category`, `/${selectedIndex}/category?q=${query}`);
```
Comment (jamessspanggg) [inline src/components/search/modules/NavSearchBar.js:9]: should we not route them for empty queries with only white spaces? i.e. `query.trim().length === 0`

File: src/components/search/modules/NavSearchBarDropdownList.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+const Container = styled.div`
+  padding-top: 10px;
+`;
+
+const ContentContainer = styled.div`
+  background: white;
+  padding: 0px 10px 8px 10px;
+`;
+
+const NavSearchBar = () => {
```
Comment (jamessspanggg) [inline src/components/search/modules/NavSearchBarDropdownList.js:24]: seems like this component is no longer being used

File: src/components/search/modules/SearchBarV2.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+          spacing: 'none',
+          direction: 'row',
+        }}
+      >
+        {isDesktop || isTablet ? null : (
+          <Stack justify="center">
+            <Popover
+              content={
+                <div>
+                  <ListChoice title="Wishes" onClick={() => setSelectedIndex('Wishes')} />
```
Comment (jamessspanggg) [inline src/components/search/modules/SearchBarV2.js:53]: should we set the `opened` field of the Popover to false when a list choice is clicked? would that result to a better UX?

File: src/components/search/modules/NavSearchBarDropdownList.js (hunk 2 above)
Comment (jamessspanggg) [reply src/components/search/modules/NavSearchBarDropdownList.js:24]: we might need to do the "Suggestion" in the future, so keeping this component for now

File: src/components/search/modules/SearchBarV2.js (hunk 3 above)
Comment (kohchihao) [reply src/components/search/modules/SearchBarV2.js:53]: Actually i tried the `opened` prop but with that prop, i lose out on the `click outside, auto close`. So i decided to not use the `opened` prop.

**Pull request merged**